### DISCUSSION TITLE
password in OAuth2 flow of kong

### DISCUSSION THREAD

**Discussion by Korakbhakta02** [Kong/kong#12563] [Help] [answered]
The data plane is not using database, for which when I Try to post credentials and password in /oauth2/tokens, this error shows up:
**[off] cannot create 'oauth2_tokens' entities when not using a database**

How to fix this?

**Comment by chronolaw** [comment, marked as answer]
oauth2 must work with DB, see: https://docs.konghq.com/hub/kong-inc/oauth2/.
